homeassistant/components/devolo_home_control/switch.py

```diff
@@ -0,0 +1,144 @@
+            self._unique_id
+        )
+        self._is_on = self._binary_switch_property.state
+
+        if hasattr(self._device_instance, "consumption_property"):
+            self._consumption = self._device_instance.consumption_property.get(
+                self._unique_id.replace("BinarySwitch", "Meter")
+            ).current
+        else:
+            self._consumption = None
+
+        self.subscriber = None
+
+    async def async_added_to_hass(self):
+        """Call when entity is added to hass."""
+        self.subscriber = async_dispatcher_connect(
+            self.hass, self._device_instance.itemName, self.update
+        )
+        self._homecontrol.publisher.register(
+            self._device_instance.uid, self.subscriber, self.update
+        )
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the switch."""
+        return self._unique_id
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "identifiers": {(DOMAIN, self._device_instance.uid)},
+            "name": self.name,
+            "manufacturer": self._brand,
+            "model": self._model,
+        }
+
+    @property
+    def device_id(self):
+        """Return the ID of this switch."""
+        return self._unique_id
+
+    @property
+    def name(self):
+        """Return the display name of this switch."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    @property
+    def is_on(self):
+        """Return the state."""
+        return self._is_on
+
+    @property
+    def current_power_w(self):
+        """Return the current consumption."""
+        return self._consumption
+
+    @property
+    def available(self):
+        """Return the online state."""
+        return self._available
+
+    def turn_on(self, **kwargs):
+        """Switch on the device."""
+        self._is_on = True
+        self._binary_switch_property.set_binary_switch(state=True)
+
+    def turn_off(self, **kwargs):
+        """Switch off the device."""
+        self._is_on = False
+        self._binary_switch_property.set_binary_switch(state=False)
+
+    @callback
+    def update(self, message=None):
+        """Update the binary switch state and consumption."""
+        if message[0].startswith("devolo.BinarySwitch"):
+            self._is_on = self._device_instance.binary_switch_property[message[0]].state
+        elif message[0].startswith("devolo.Meter"):
+            self._consumption = self._device_instance.consumption_property[
+                message[0]
+            ].current
+        elif message[0].startswith("hdm"):
+            self._available = self._device_instance.is_online()
+        else:
+            _LOGGER.debug("No valid message received")
+            _LOGGER.debug(message)
+        self.async_schedule_update_ha_state()
```

Comment:
If this is run in another thread than the event loop we need to use the sync version.
```suggestion
        self.schedule_update_ha_state()
```

Reply:
Use self.schedule_update_ha_state()  now